src/main/java/com/hedvig/backoffice/graphql/resolvers/ClaimResolver.kt

```diff
@@ -0,0 +1,134 @@
+package com.hedvig.backoffice.graphql.resolvers
+
+import com.coxautodev.graphql.tools.GraphQLResolver
+import com.hedvig.backoffice.graphql.Util
+import com.hedvig.backoffice.graphql.dataloaders.MemberLoader
+import com.hedvig.backoffice.graphql.types.Claim
+import com.hedvig.backoffice.graphql.types.ClaimFileUpload
+import com.hedvig.backoffice.graphql.types.ClaimTypes
+import com.hedvig.backoffice.graphql.types.Member
+import com.hedvig.backoffice.graphql.types.claims.*
+import com.hedvig.backoffice.services.UploadedFilePostprocessor
+import com.hedvig.backoffice.services.product_pricing.ProductPricingService
+import com.hedvig.backoffice.services.product_pricing.dto.contract.Contract
+import graphql.schema.DataFetchingEnvironment
+import org.springframework.stereotype.Component
+import java.util.*
+import java.util.concurrent.CompletableFuture
+import java.util.stream.Collectors
+
+@Component
+class ClaimResolver(
+    private val memberLoader: MemberLoader,
+    private val uploadedFilePostprocessor: UploadedFilePostprocessor,
+    private val productPricingService: ProductPricingService
+) : GraphQLResolver<Claim?> {
+    fun getMember(claim: Claim): CompletableFuture<Member> {
+        return try {
+            memberLoader.load(claim.memberId)
+        } catch (exception: Exception) {
+            CompletableFuture()
+        }
+    }
+
+    fun getClaimFiles(claim: Claim): List<ClaimFileUpload> {
+        val claimFileUploads: MutableList<ClaimFileUpload> = ArrayList()
+        val claimFiles = claim.claimFiles
+        if (claimFiles.isEmpty()) {
+            return claimFileUploads
+        }
+        val claimFileUploadDtos = claimFiles.stream().map { (claimFileId, bucket, key, claimId, contentType, UploadedAt, _, category) ->
+            val claimUpload = ClaimFileUpload(
+                claimFileId,
+                uploadedFilePostprocessor.processFileUrl(key, bucket),
+                UploadedAt,
+                claimId,
+                category,
+                contentType
+            )
+            claimUpload
+        }.collect(Collectors.toList())
+        claimFileUploads.addAll(claimFileUploadDtos)
+        return claimFileUploads
+    }
```

Comment:
Can make this one a bit neater
```suggestion
    fun getClaimFiles(claim: Claim): List<ClaimFileUpload> {
        if (claimFiles.isEmpty()) {
            return emptyList()
        }
        return claimFiles.stream().map { claimFile ->
            ClaimFileUpload(
                claimFile.claimFileId,
                uploadedFilePostprocessor.processFileUrl(claimFile.key, claimFile.bucket),
                claimFile.UploadedAt,
                claimFile.claimId,
                claimFile.category,
                claimFile.contentType
            )
        }
    }
```

Reply:
Nice one!